runtime/src/main/kotlin/kotlin/native/concurrent/Worker.kt

```diff
@@ -103,17 +107,45 @@ public inline class Worker @PublishedApi internal constructor(val id: Int) {
     }
 
     /**
-     * Process pending job(s) on the queue of this worker, returns `true` if something was processed
-     * and `false` otherwise. Note that jobs scheduled with [executeAfter] using non-zero timeout are
+     * Process pending job(s) on the queue of this worker.
+     * Note that jobs scheduled with [executeAfter] using non-zero timeout are
      * not processed this way. If termination request arrives while processing the queue via this API,
      * worker is marked as terminated and will exit once the current request is done with.
+     *
+     * @throws [IllegalArgumentException] if this request is executed on non-current [Worker].
+     * @return `true` if request(s) was processed and `false` otherwise.
      */
     public fun processQueue(): Boolean = processQueueInternal(id)
 
+    /**
+     * Park execution of the current worker until a new request arrives or timeout specified in
+     * [timeoutMicroseconds] elapsed. If [processQueue] is true, pending queue elements are processed the same way,
+     * as specified in [processQueue] function.
+     *
+     * @param timeoutMicroseconds defines how long to park worker if no requests arrive.
```

Comment:
Can it be nanoseconds, as requested in [KT-34570](https://youtrack.jetbrains.com/issue/KT-34570)?

Reply:
It could, but not sure if nanoseconds are proper granularity for those APIs. Can ask @elizarov on that.

Reply:
I’d prefer nanoseconds here. It would be easier to port code between JVM and native then.

Reply:
Also a question ob why it is an instance method of Worker and what happens if I call it from another worker? Which one gets parked?

Reply:
* @throws [IllegalStateException] if this request is executed on non-current [Worker].

Reply:
OK, can switch to nanoseconds, although it doesn't look entirely correct.

Reply:
No, I will avoid using microseconds, otherwise API gets really crazy when `executeAfter` takes microseonds and this one nanoseconds.